### PR TITLE
Add fly pg failover command

### PR DESCRIPTION
**Pull request by rugwirobaker**
- Add fly pg failover command

### CODE REVIEW

File: internal/command/postgres/failover.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		return fmt.Errorf("list of machines could not be retrieved: %w", err)
+	}
+	ctx = flaps.NewContext(ctx, flapsClient)
+
+	machines, err := flapsClient.List(ctx, "started")
+	if err != nil {
+		return fmt.Errorf("machines could not be retrieved %w", err)
+	}
+
+	leader, err := fetchPGLeader(ctx, machines)
```
Comment (davissp14) [inline internal/command/postgres/failover.go:78]: This is fine as is, but anything that results in a `stolonctl` call can be  issued from any member in the cluster.

File: internal/command/postgres/failover.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+	app, err := client.GetAppCompact(ctx, appName)
+	if err != nil {
+		return fmt.Errorf("get app: %w", err)
+	}
+
+	if !app.IsPostgresApp() {
+		return fmt.Errorf("app %s is not a Postgres app", app.Name)
+	}
+
```
Comment (davissp14) [inline internal/command/postgres/failover.go:55]: We may want to toss an error if the App is on Nomad.

File: internal/command/postgres/failover.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	if err != nil {
+		return fmt.Errorf("list of machines could not be retrieved: %w", err)
+	}
+	ctx = flaps.NewContext(ctx, flapsClient)
+
+	machines, err := flapsClient.List(ctx, "started")
+	if err != nil {
+		return fmt.Errorf("machines could not be retrieved %w", err)
+	}
+
```
Comment (davissp14) [inline internal/command/postgres/failover.go:77]: We should look to acquire a cluster wide lease before issuing the failover.

File: internal/command/postgres/failover.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+		return fmt.Errorf("failover is not available for standalone postgres")
+	}
+
+	pgclient := flypg.New(appName, dialer)
+
+	fmt.Fprintf(io.Out, "Performing a failover\n")
+	if err := pgclient.Failover(ctx); err != nil {
+		return fmt.Errorf("failed to trigger failover %w", err)
+	}
+
```
Comment (davissp14) [inline internal/command/postgres/failover.go:98]: What are your thoughts on waiting on health checks here?

**Review by davissp14 — CHANGES_REQUESTED** [review]
Great start!  Just a few comments.

File: internal/command/postgres/failover.go (hunk 1 above)
Comment (rugwirobaker) [reply internal/command/postgres/failover.go:78]: I'm only fetching the leader to check the image version It's running. That's going to change in https://github.com/superfly/flyctl/pull/1315

**Review by davissp14 — APPROVED** [review]
LGTM!

File: internal/command/postgres/failover.go [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+	if err != nil {
+		return fmt.Errorf("machines could not be retrieved %w", err)
+	}
+
+	if err := hasRequiredVersionOnMachines(machines, MinPostgresHaVersion, MinPostgresHaVersion); err != nil {
+		return err
+	}
+
+	// acquire cluster wide lock
+	for _, machine := range machines {
```
Comment (davissp14) [inline internal/command/postgres/failover.go:88]: We should perform this after the single node check below.

**Pull request merged**